### DISCUSSION TITLE
Creating a library that uses vcpkg libraries but doesn't link to them

### DISCUSSION THREAD

**Discussion by grisharav** [microsoft/vcpkg#41833] [Q&A] [answered]
Hi,
We are using vcpkg to create a static library, lets call it LIB. The lib itself uses and provides an API that contains code from vcpkg provided packages e.g. `fmt` and others.
When I define the library purely as:
```CMAKE
add_library(LIB ${LIBS_SRC})
```
The generated compilation commands don't add an include to the packages folder, so the compilation obviously fails. Only if I add:
```
target_link_libraries(LIB fmt::fmt)
```
The additional include directories are properly generated and the library compiles. Is this the recommended way to do this? Basically I would only like to add include directory to the vcpkg include folder, how do I achieve that?

**Comment by Osyotr** [comment, marked as answer]
> Is this the recommended way to do this?

Yes, using exported targets is the recommended way to consume packages. 
You might want to specify scope though. https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#target-command-scope

**Reply by grisharav** [reply]
Thanks for the reply @Osyotr .
So basically you are saying that when creating libraries (static libs) they should always be linked in with their dependencies?

**Reply by Osyotr** [reply]
If it's a compile time dependency, yes.
Note that you are linking targets. This may or may not involve linking actual lib/a/so/dylib files.
